pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go

```diff
@@ -345,136 +534,226 @@ func (lbaas *LbaasV2) EnsureLoadBalancer(clusterName string, apiService *api.Ser
 		return nil, fmt.Errorf("error checking if openstack load balancer already exists: %v", err)
 	}
 
-	// TODO: Implement a more efficient update strategy for common changes than delete & create
-	// In particular, if we implement hosts update, we can get rid of UpdateHosts
-	if exists {
-		err := lbaas.EnsureLoadBalancerDeleted(clusterName, apiService)
+
+	var loadbalancer *loadbalancers.LoadBalancer
+	name := cloudprovider.GetLoadBalancerName(apiService)
+
+	if !exists {
+		glog.V(2).Infof("Creating loadbalancer %s", name)
+		loadbalancer, err = lbaas.createLoadBalancer(apiService, name)
 		if err != nil {
-			return nil, fmt.Errorf("error deleting existing openstack load balancer: %v", err)
+			// cleanup what was created so far
+			_ = lbaas.EnsureLoadBalancerDeleted(clusterName, apiService)
+			return nil, fmt.Errorf("Error creating loadbalancer %s: %v", name, err)
+		}
```

Comment:
aha.  Nope, I just mis-read which function you were calling.  Nothing to see here, carry on ;)